Add "From Zero to Subproject Lead" article (#388)
kubernetes/contributor-site: content/en/blog/2023/from-zero-to-subproject-lead.md

---
layout: blog
title: "From Zero to Kubernets Subproject Lead"
date: 2023-03-29
slug: from-zero-to-k8s-subproject-lead
author: "Ala Dewberry (VMware)"
---

Getting started in any open-source community can be daunting, especially if it’s a big one like

> Kubernetes. I wanted to write this post to share my experience and encourage others to join up. All

Reviewer: ```suggestion
Kubernetes. I wrote this post to share my experience and encourage others to join up. All
```

it takes is some curiosity and a willingness to show up!

Here’s how my journey unfolded at a high level:

1. What am I interested in? Is there a SIG (Special Interest Group) or a WG (Working Group) that is
   dedicated to that topic, or something similar? 

> 2. Sign up for their mailing list and start hopping on meetings!

Reviewer: ```suggestion
2. Sign up for their mailing list and start hopping on meetings.
```

3. When (never if!) there are opportunities to help out and it aligns with your skills and desired

>    growth areas, raise your hand!

Reviewer: ```suggestion
   growth areas, raise your hand.
```

4. Ask for lots of help and don’t be shy about not knowing everything (or anything!)

> 5. Keep plugging along, even if progress isn’t as fast as you would like it to be

Reviewer: ```suggestion
5. Keep plugging along, even if progress isn’t as fast as you would like it to be.
```

## Starting up

First things first. What are you interested in learning more about? There are so many wonderful SIGs
and working groups in the Kubernetes community: there’s something for everyone. And continuing to
show up and participate will be so much easier if you think what you are doing is
interesting. Likewise, continued participation is what keeps the community thriving, so that
interest will drive you to have more of an impact. 

> The following is also essential - it’s ok to show up knowing nothing! I remember showing up knowing

Reviewer: ```suggestion
Also: it’s ok to show up knowing nothing! I remember showing up knowing
```

very little about Kubernetes or how the community itself worked. And while I know more about how the
community functions today, I am still learning all the time about it and the project. Fortunately,
the community is full of friendly people who want to help you learn. Learning as you go is expected
and celebrated. When you raise your hand to do something, even if you know nothing, people will
cheer and help you along the way. 

This method was my exact story. It was my first or second meeting with [SIG
Security](https://github.com/kubernetes/community/tree/master/sig-security), and [Pushkar
Joglekar](https://github.com/PushkarJ) mentioned that he needed a lead for a subproject he was

> creating after having done a security assessment of CAPI. Everyone was so friendly in the meeting

Reviewer 2: Optionally:
```suggestion
creating after having done a security assessment of [Cluster API](https://cluster-api.sigs.k8s.io/).
Everyone was so friendly in the meeting
```

> that I thought - hey, why not try it out? And since then, I have received so much support and

Reviewer: ```suggestion
that I thought, "Hey, why not try it out?" And since then, I have received so much support and
```

> encouragement from my co-leads who are delighted to have me, _especially_ because I am a beginner:

Reviewer: ```suggestion
encouragement from my co-leads who are delighted to have me, _especially_ because I am a beginner;
```

new participation is what keeps the community healthy.

## Always learning

My participation has also been a great learning experience on several fronts. First, I have been
exposed to techniques for how to build community consensus. It’s simple stuff: show up at other SIG
or working group meetings, share your ideas or where you are looking for help, find people who are
interested and have the knowledge to help, build an action plan together, do it, and share as you
execute. But the other thing that I’m learning is that building this consensus and executing it in a
transparent, inviting way simply takes time. 

I also have to be patient with myself and remember that I am learning as I go. The [Kubernetes git
repo](https://github.com/kubernetes/kubernetes) can be daunting to navigate. Knowing the next best

> step isn’t always obvious. But this is where my third learning curve - how to engage the community
> to get what I need - comes into play. It turns out that asking questions in the [Kubernetes Slack

Reviewer: ```suggestion
step isn’t always obvious. But this is where my third learning curve, how to engage the community
to get what I need, comes into play. It turns out that asking questions in the [Kubernetes Slack
```

Could also use m-dashes.

workspace](https://slack.k8s.io/) and bringing my topics to the SIG Security meetings when I need
help is an amazing way to get what I need! Again, simple stuff, but until you do it, it’s not always
obvious.

## Why you - a beginner - are important to the project

In many ways, beginners are the most important part of the community. To put a finer point on it:
asking for, receiving, and then giving help is a very relevant part of how the community grows and
flourishes. When we take on and then pass on knowledge, we ensure that the community grows enough to
keep supporting the needs of the people who rely on the project, whatever it is. You have

> superpowers as a beginner! 

Reviewer 2: If you want to, emphasize that beginners can see problems (eg a gap in the docs) that experienced contributors find hard to see. It really is a superpower.

Reviewer 2: nit on suggestion: “truly”

transparent, inviting way simply takes time. 

I also have to be patient with myself and remember that I am learning as I go. The [Kubernetes git
repo](https://github.com/kubernetes/kubernetes) can be daunting to navigate. Knowing the next best
step isn’t always obvious. But this is where my third learning curve, how to engage the community
to get what I need, comes into play. It turns out that asking questions in the [Kubernetes Slack
workspace](https://slack.k8s.io/) and bringing my topics to the SIG Security meetings when I need
help is an amazing way to get what I need! Again, simple stuff, but until you do it, it’s not always
obvious.

## Why you - a beginner - are important to the project

In many ways, beginners are the most important part of the community. To put a finer point on it:
asking for, receiving, and then giving help is a very relevant part of how the community grows and
flourishes. When we take on and then pass on knowledge, we ensure that the community grows enough to
keep supporting the needs of the people who rely on the project, whatever it is. You have
superpowers as a beginner! 

> I hope people who read this post have their curiosity peaked about getting involved in the

Reviewer 2: ```suggestion
I hope people who read this post have their curiosity piqued about getting involved in the
```

Author: Good one - I'll open a new PR with it!
